tools/slicec-cs/src/decoding.rs

```diff
@@ -239,7 +239,7 @@ pub fn decode_sequence(sequence_ref: &TypeRef<Sequence>, namespace: &str, encodi
         write!(
             code,
             "({}[])",
-            element_type.cs_type_string(namespace, TypeContext::Nested, true),
+            element_type.cs_type_string(namespace, TypeContext::Nested, false),
```

Comment:
This bool sets `ignore_optional`.

Before, it was true, so the cast type would be missing a `?` when it needed one.
Changing this to `false` means we add the `?` for optional types now.

Reply:
The inconsistency between sequences and dictionaries is odd. I'd rather use a cast for both.